dynamic_env_pytest_tests/tests/services/s3_gate/test_s3_gate.py

```diff
@@ -0,0 +1,559 @@
+
+        with allure.step(f"Delete bucket {bucket_1}"):
+            s3_gate_bucket.delete_bucket_s3(self.s3_client, bucket_1)
+            self.tick_epochs_and_wait(1)
+
+        with allure.step(f"Check bucket {bucket_1} deleted"):
+            with pytest.raises(Exception, match=r".*Not Found.*"):
+                s3_gate_bucket.head_bucket(self.s3_client, bucket_1)
+
+    @allure.title("Test S3 Object API")
+    @pytest.mark.parametrize(
+        "file_type", ["simple", "large"], ids=["Simple object", "Large object"]
+    )
+    def test_s3_api_object(self, file_type, two_buckets, simple_object_size, complex_object_size):
+        """
+        Test base S3 Object API (Put/Head/List) for simple and large objects.
+        """
+        file_path = generate_file(
+            simple_object_size if file_type == "simple" else complex_object_size
+        )
+        file_name = self.object_key_from_file_path(file_path)
+
+        bucket_1, bucket_2 = two_buckets
+
+        for bucket in (bucket_1, bucket_2):
+            with allure.step("Bucket must be empty"):
+                objects_list = s3_gate_object.list_objects_s3(self.s3_client, bucket)
+                assert not objects_list, f"Expected empty bucket, got {objects_list}"
+
+            s3_gate_object.put_object_s3(self.s3_client, bucket, file_path)
+            s3_gate_object.head_object_s3(self.s3_client, bucket, file_name)
+
+            bucket_objects = s3_gate_object.list_objects_s3(self.s3_client, bucket)
+            assert (
+                file_name in bucket_objects
+            ), f"Expected file {file_name} in objects list {bucket_objects}"
+
+        with allure.step("Check object's attributes"):
+            for attrs in (["ETag"], ["ObjectSize", "StorageClass"]):
+                s3_gate_object.get_object_attributes(self.s3_client, bucket, file_name, *attrs)
+
+    @allure.title("Test S3 Sync directory")
+    @pytest.mark.aws_cli_only
+    def test_s3_sync_dir(self, bucket, simple_object_size):
+        """
+        Test checks sync directory with AWS CLI utility.
+        """
+        file_path_1 = os.path.join(os.getcwd(), ASSETS_DIR, "test_sync", "test_file_1")
+        file_path_2 = os.path.join(os.getcwd(), ASSETS_DIR, "test_sync", "test_file_2")
+        key_to_path = {"test_file_1": file_path_1, "test_file_2": file_path_2}
+
+
+        generate_file_with_content(simple_object_size, file_path=file_path_1)
+        generate_file_with_content(simple_object_size, file_path=file_path_2)
+
+        self.s3_client.sync(bucket_name=bucket, dir_path=os.path.dirname(file_path_1))
+
+        with allure.step("Check objects are synced"):
+            objects = s3_gate_object.list_objects_s3(self.s3_client, bucket)
+
+        with allure.step("Check these are the same objects"):
+            assert set(key_to_path.keys()) == set(
+                objects
+            ), f"Expected all objects saved. Got {objects}"
```

Comment:
it seems to me that will we see this failure also then `objects` set is wider then `key_to_path.keys()`

if so, i'd rephrase
```suggestion
            ), f"Expected exact objects saved. Got {objects}"
```

Reply:
https://github.com/nspcc-dev/neofs-testcases/pull/703